js/content/store.js

```diff
@@ -813,45 +813,47 @@ class AppPageClass extends StorePageClass {
 
-        // there is no add to wishlist button and game is not purchased yet, add required nodes
-        if (!document.querySelector("#add_to_wishlist_area") && !document.querySelector(".game_area_already_owned")) {
+        // If game is already wishlisted, add required nodes
+        if (!document.getElementById("add_to_wishlist_area")) {
             let firstButton = document.querySelector(".queue_actions_ctn a.queue_btn_active");
-            HTML.beforeEnd(firstButton, "<div id='add_to_wishlist_area_success' style='display: inline-block;'></div>");
+            HTML.beforeBegin(firstButton, '<div id="add_to_wishlist_area_success"></div>');
 
-            let wlSuccessArea = document.querySelector("#add_to_wishlist_area_success");
+            let wlSuccessArea = document.getElementById("add_to_wishlist_area_success");
             DOMHelper.wrap(wlSuccessArea, firstButton);
-            HTML.beforeBegin(wlSuccessArea,  `<div id='add_to_wishlist_area' style='display: none;'><a class='btnv6_blue_hoverfade btn_medium' href='javascript:AddToWishlist(${this.appid}, \\"add_to_wishlist_area\\", \\"add_to_wishlist_area_success\\", \\"add_to_wishlist_area_fail\\", \\"1_5_9__407\\" );'><span>${Localization.str.add_to_wishlist}</span></a></div>`);
-            HTML.beforeBegin(wlSuccessArea, `<div id='add_to_wishlist_area_fail' style='display: none;'></div>`);
+            HTML.beforeBegin(wlSuccessArea,
+                `<div id="add_to_wishlist_area" style="display: none;">
+                    <a class="btnv6_blue_hoverfade btn_medium" data-tooltip-text="${Localization.str.add_to_wishlist_tooltip}">
+                        <span>${Localization.str.add_to_wishlist}</span>
+                    </a>
+                </div>
+                <div id="add_to_wishlist_area_fail" style="display: none;">
+                    <b>${Localization.str.error}</b>
+                </div>`);
         }
 
-        let addBtn = document.querySelector("#add_to_wishlist_area > a");
-        if (addBtn && !addBtn.href) addBtn.href = `javascript:AddToWishlist( ${this.appid}, 'add_to_wishlist_area', 'add_to_wishlist_area_success', 'add_to_wishlist_area_fail', null, 'add_to_wishlist_area2' );`
-
-        let successNode = document.querySelector("#add_to_wishlist_area_success");
-        if (!successNode) { return; }
+        let addBtn = document.getElementById("add_to_wishlist_area");
+        let successBtn = document.getElementById("add_to_wishlist_area_success");
 
-        let innerNode = successNode.querySelector("[data-tooltip-text]");
-        if (!innerNode) { return; }
-        innerNode.dataset.tooltipText = Localization.str.remove_from_wishlist;
+        if (!addBtn.querySelector("a").href) {
+            addBtn.querySelector("a").href = `javascript:AddToWishlist( ${this.appid}, 'add_to_wishlist_area', 'add_to_wishlist_area_success', 'add_to_wishlist_area_fail', null, 'add_to_wishlist_area2' );`;
+        }
```

Comment:
Can't you remove this check and put the line into the if-clause above?

Reply:
Also, I wouldn't call `querySelector` twice. In this case I think it would be better to make another variable.